ambry-router/src/main/java/com.github.ambry.router/SimpleOperationTracker.java

```diff
@@ -72,25 +74,43 @@
   /**
    * Constructor for an {@code SimpleOperationTracker}.
    *
-   * @param datacenterName The datacenter where the router is located.
+   * @param routerConfig The {@link RouterConfig} containing the configs for operation tracker.
+   * @param operationClass The operation class in which operation tracker is used.
    * @param partitionId The partition on which the operation is performed.
-   * @param crossColoEnabled {@code true} if requests can be sent to remote replicas, {@code false}
-   *                                otherwise.
    * @param originatingDcName The original DC where blob was put.
-   * @param includeNonOriginatingDcReplicas if take the option to include remote non originating DC replicas.
-   * @param replicasRequired The number of replicas required for the operation.
-   * @param successTarget The number of successful responses required to succeed the operation.
-   * @param parallelism The maximum number of inflight requests at any point of time.
    * @param shuffleReplicas Indicates if the replicas need to be shuffled.
    */
-  SimpleOperationTracker(String datacenterName, PartitionId partitionId, boolean crossColoEnabled,
-      String originatingDcName, boolean includeNonOriginatingDcReplicas, int replicasRequired, int successTarget,
-      int parallelism, boolean shuffleReplicas) {
+  SimpleOperationTracker(RouterConfig routerConfig, Class operationClass, PartitionId partitionId,
+      String originatingDcName, boolean shuffleReplicas) {
+    // populate tracker parameters based on operation type
+    boolean crossColoEnabled = false;
+    boolean includeNonOriginatingDcReplicas = true;
+    int replicasRequired = Integer.MAX_VALUE;
+    if (GetOperation.class.isAssignableFrom(operationClass)) {
+      successTarget = routerConfig.routerGetSuccessTarget;
+      parallelism = routerConfig.routerGetRequestParallelism;
+      crossColoEnabled = routerConfig.routerGetCrossDcEnabled;
+      includeNonOriginatingDcReplicas = routerConfig.routerGetIncludeNonOriginatingDcReplicas;
+      replicasRequired = routerConfig.routerGetReplicasRequired;
+    } else if (operationClass == PutOperation.class) {
```

Comment:
One other option is to make an enum to represent the operation tracker types (i.e. GET, PUT, DELETE, TTL_UPDATE). This is a little more explicit and resilient to new classes/name changes.